gravitee-apim-gateway/gravitee-apim-gateway-policy/src/main/java/io/gravitee/gateway/policy/impl/PolicyFactoryImpl.java

```diff
@@ -62,9 +63,13 @@ public Policy create(
         }
 
         final ExecutablePolicy executablePolicy = new ExecutablePolicy(policyManifest.id(), policy, headMethod, streamMethod);
-        final String condition = policyMetadata.getCondition();
-        if (condition != null && !condition.isBlank()) {
-            return new ConditionalExecutablePolicy(executablePolicy, condition, conditionEvaluator);
+
+        if (conditionEvaluator != null && !Objects.equals(true, policyMetadata.metadata().get(PolicyMetadata.MetadataKeys.JUPITER))) {
```

Comment:
I don't think this is necessary because in context of v3, you never obtain a jupiter policy otherwise the code above won't work (the one searching about method annotation)

Reply:
Not sure I get your point. This factory can be called in different situations:

- Jupiter disabled: we are executing in V3 mode without jupiter. When a condition is defined the `ExecutablePolicy` become a `ConditionalExecutablePolicy`. 
- Jupiter enabled with v3 mode: we a in v3 compatibility mode so with keep the same behavior as the v3.
- Jupiter enabled with jupiter mode: here Jupiter handle the condition by itself. We are using the same policy factory because we are able to execute v3 policy (thanks to adapter). We don't want to execute condition twice (1 by jupiter, 1 by ConditionalExecutablePolicy). This allows to avoid creating a `ConditionExecutablePPolicy` when Jupiter mode is enabled.

Makes sense?

Reply:
Actually while using jupiter mode we use this factory only when the policy is not a jupiter policy, I think it is more about is jupiter enabled or not than a jupiter policy. When jupiter is disabled the ConditionalExecutablePolicy must be enabled if required. From my point of view, the policy metadata should have dedicated attribute about jupiter enablement.

Reply:
As explained above there are 2 cases to distinguish:

- Jupiter enabled with v3 mode: we a in v3 compatibility mode so with keep the same behavior as the v3.
- Jupiter enabled with jupiter mode: here **Jupiter handle the condition by itself**. We are using the same policy factory because we are able to execute v3 policy (thanks to adapter). We don't want to execute condition twice (1 by jupiter, 1 by `ConditionalExecutablePolicy`). This allows to avoid creating a `ConditionExecutablePolicy` when Jupiter execution mode is active (because it's managed by jupiter itself).